packages/core/src/controllers/ConnectionController.ts

```diff
@@ -86,12 +87,46 @@ export const ConnectionController = {
   async connectWalletConnect() {
     StorageUtil.setConnectedConnector('WALLET_CONNECT')
 
-    await ChainController.state?.universalAdapter?.connectionControllerClient?.connectWalletConnect?.(
-      uri => {
-        state.wcUri = uri
-        state.wcPairingExpiry = CoreHelperUtil.getPairingExpiry()
+    if (CoreHelperUtil.isTelegram()) {
```

Comment:
Can we merge this with regular flow as well? I mean instead of separating them as Telegram or not, would it be possible to use same approach for both cases?

Reply:
https://walletconnect.slack.com/archives/C03RVH94K5K/p1727854874171529?thread_ts=1727850585.866219&cid=C03RVH94K5K